lib/chef/provider_resolver.rb

```diff
@@ -68,27 +68,39 @@ def provided_by?(provider_class)
       potential_handlers.include?(provider_class)
     end
 
-    def self.includes_handler?(resource_name, provider_class)
-      priority_map.includes_handler?(resource_name, provider_class)
-    end
-
     def enabled_handlers
-      @enabled_handlers ||=
-        potential_handlers.select { |handler| !overrode_provides?(handler) || handler.provides?(node, resource) }
+      potential_handlers.select { |handler| !overrode_provides?(handler) || handler.provides?(node, resource) }
```

Comment:
It's not too big a deal as a rule, but in general gc'ing runtimes can
optimize better when things remain entirely on the stack.

On Wed, Jul 1, 2015, 5:17 PM Lamont Granquist <notifications@github.com>
wrote:

> In lib/chef/provider_resolver.rb
> <https://github.com/chef/chef/pull/3627#discussion_r33735005>:
>
> >      def enabled_handlers
> > -      @enabled_handlers ||=
> > -        potential_handlers.select { |handler| !overrode_provides?(handler) || handler.provides?(node, resource) }
> > +      potential_handlers.select { |handler| !overrode_provides?(handler) || handler.provides?(node, resource) }
>
> Hmmm, I typically like to do it when i can so that in later refactorings I
> can just re-use the method and not have to worry about performance...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/chef/chef/pull/3627/files#r33735005>.
>